### PR TITLE
Features/manta recipe

### PR DESCRIPTION
**Pull request by northwestwitch**
Added Manta recipe and remove redundant code from mip.pl. Tested on rasta.

### CODE REVIEW

File: lib/MIP/Recipes/Manta.pm [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    my $outfile_prefix = $family_id . $outfile_tag . $call_type;
+    my $outfile_path_prefix = catfile( $temp_directory, $outfile_prefix );
+
+    ## Assign suffix
+    my $infile_suffix = get_file_suffix(
+        {
+            parameter_href => $parameter_href,
+            suffix_key     => q{alignment_file_suffix},
+            jobid_chain    => $parameter_href->{pgatk_baserecalibration}{chain}
+            ,    #Get infile_suffix from baserecalibration jobid chain
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: Remove #Get inf... comment

File: lib/MIP/Recipes/Manta.pm [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+                FILEHANDLE            => $FILEHANDLE,
+            }
+        );
+
+        ## Assign directories
+        my $insample_directory = catdir( $active_parameter_href->{outdata_dir},
+            $sample_id, $outaligner_dir );
+
+        ## Add merged infile name after merging all BAM files per sample_id
+        my $infile = $file_info_href->{$sample_id}{merge_infile};    #Alias
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: Move # Alias to line above

File: lib/MIP/Recipes/Manta.pm [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+        ## Assign file_tags
+        my $infile_tag =
+          $file_info_href->{$sample_id}{pgatk_baserecalibration}{file_tag};
+        my $infile_prefix = $infile . $infile_tag;
+
+        my $infile_path = catfile( $insample_directory,
+            $infile_prefix . substr( $infile_suffix, 0, 2 ) . q{*} );
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: q{*} -> $ASTERIX

File: lib/MIP/Recipes/Manta.pm [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+
+    ## Assemble file paths by adding file ending
+    my @file_paths = map { $file_path_prefix{$_} . $infile_suffix }
+      @{ $active_parameter_href->{sample_ids} };
+
+    manta_config(
+        {
+            infile_paths_ref  => \@file_paths,
+            outdirectory_path => $temp_directory,
+            referencefile_path =>
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: We should add this '$active_parameter_href->{human_genome_reference}' as a default to the sub as $human_genome_reference just as temp_directory

File: lib/MIP/Recipes/Manta.pm [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+        }
+    );
+    say {$FILEHANDLE} $NEWLINE;
+
+    ## Perl wrapper for writing gzip recipe to $FILEHANDLE
+    gzip(
+        {
+            decompress  => 1,
+            stdout      => 1,
+            infile_path => catfile(
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: Assemble infile_path before call in my $infile_path_gzip = catfile...

File: lib/MIP/Recipes/Manta.pm [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+            FILEHANDLE   => $FILEHANDLE,
+        }
+    );
+    say {$FILEHANDLE} $NEWLINE;
+
+    ## Copies file from temporary directory.
+    say {$FILEHANDLE} q{## Copy file from temporary directory};
+    migrate_file(
+        {
+            infile_path  => $outfile_path_prefix . $outfile_suffix . q{*},
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: $ASTERIX

File: mip.pl [hunk 7]
```diff
@@ -1963,7 +1963,16 @@ BEGIN
...
 
-    $log->info("[Manta]\n");
+    $log->info( q{[Manta]} . $NEWLINE );
+    my $program_name = lc q{manta};
+
+    my $outfamily_directory = catfile(
+        $active_parameter{outdata_dir},
+        $active_parameter{family_id},
+        $active_parameter{outaligner_dir},
+        $active_parameter{outdir_name},
```
Comment (henrikstranneheim) [inline mip.pl]: What value has this one: $active_parameter{outdir_name}

File: lib/MIP/Recipes/Manta.pm (hunk 4 above)
Comment (northwestwitch) [reply lib/MIP/Recipes/Manta.pm]: You want to pass it from mip.pl ?

File: lib/MIP/Recipes/Manta.pm [hunk 8]
```diff
@@ -31,14 +31,15 @@ BEGIN {
...
 Readonly my $NEWLINE    => qq{\n};
+Readonly my $ASTERISK   => q{*};
 
 sub analysis_manta {
 
 ## analysis_manta
 
 ## Function : Joint analysis of structural variation
 ## Returns  : ""
-## Arguments: $parameter_href, $active_parameter_href, $sample_info_href, $file_info_href, $infile_lane_prefix_href, $job_id_href, $program_name, $family_id, $temp_directory, $outaligner_dir, $call_type, $outfamily_directory
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: Remove reference_dir sine it is not used

File: lib/MIP/Recipes/Manta.pm [hunk 9]
```diff
@@ -252,16 +260,16 @@ sub analysis_manta {
         my $insample_directory = catdir( $active_parameter_href->{outdata_dir},
             $sample_id, $outaligner_dir );
 
-        ## Add merged infile name after merging all BAM files per sample_id
-        my $infile = $file_info_href->{$sample_id}{merge_infile};    #Alias
+        ## #Alias. Add merged infile name after merging all BAM files per sample_id
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Manta.pm]: ## #Alias. Add merged infile name after merging all BAM files per sample_id ->
## Add merged infile name after merging all BAM files per sample_id

**Comment by northwestwitch** [comment]
I've fixed these 2 things

**Comment by henrikstranneheim** [comment]
Nice!
👍

**Pull request merged**